### PR TITLE
chore: update typescript-estree to new package name

### PR DESCRIPTION
**Pull request by JamesHenry**
- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If not an internal change) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by JamesHenry** [comment]
I will mark the original npm package as deprecated for clarity once this goes in

**Comment by JamesHenry** [comment]
Importantly - **`typescript-estree` will very much continue to be an agnostic TypeScript -> ESTree AST tool.**

Prettier's use case is just as important to me as ESLint's.

It's movement into the typescript-eslint monorepo is purely to help with the maintenance burden and coordinating releases across the eco-system.

**Comment by JamesHenry** [comment]
If anybody else from the Prettier Team wants to be directly involved on the monorepo please just let me know

**Comment by existentialism** [comment]
@JamesHenry happy to help!

File: website/versioned_docs/version-stable/options.md [hunk 1]
```diff
@@ -184,7 +184,7 @@ Valid options:
 - `"babel"` (via [@babel/parser](https://github.com/babel/babel/tree/master/packages/babel-parser)) _Named `"babylon"` until v1.16.0_
 - `"babel-flow"` (Same as `"babel"` but enables Flow parsing explicitly to avoid ambiguity) _First available in v1.16.0_
 - `"flow"` (via [flow-parser](https://github.com/facebook/flow/tree/master/src/parser))
-- `"typescript"` (via [typescript-estree](https://github.com/JamesHenry/typescript-estree)) _First available in v1.4.0_
+- `"typescript"` (via [@typescript-eslint/typescript-estree](https://github.com/typescript-eslint/typescript-eslint)) _First available in v1.4.0_
```
Comment (j-f1) [inline website/versioned_docs/version-stable/options.md]: Can you revert these changes please? This version of the docs will be automatically updated after the next release, when we’ll actually be using the new package.

**Comment by JamesHenry** [comment]
@existentialism Thanks! Added you to https://github.com/orgs/typescript-eslint/teams/prettier-team
